### ISSUE TITLE
cosmiconfig.search can't find file on windows, but load is working

### ISSUE THREAD

**Issue by shaobeichen** [cosmiconfig/cosmiconfig#278]
this is the problem found when using the library

https://github.com/Exlint/cz-vinyl/issues/71

cosmiconfig: "7.0.1"

```
  System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz
    Memory: 6.88 GB / 15.88 GB
  Binaries:
    Node: 16.13.1 - D:\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD       
    npm: 8.1.2 - D:\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (104.0.1293.70)
    Internet Explorer: 11.0.19041.1566
```

**Comment by d-fischer** [comment]
Absolute paths don't work in searchPlaces at the moment. You may want to check out #152 which will bring better support for global configuration (in a standards compliant directory, not directly in the home directory), but currently lacks test automation.

**Comment by 13OnTheCode** [comment]
@d-fischer 

I have the same confusion as well.

Preliminary tests suggest that the reason might be due to the unified directory structure in macOS, where there is a single home directory serving as the user's personal file space. On the other hand, Windows uses the concept of partitions and follows a layered directory structure, organizing files and directories using logical disks such as C, D, E, etc., each with its own root directory. Upon briefly examining the source code of cosmiconfig, it seems that no additional handling for Windows systems has been implemented. It is unclear whether this was intentionally done by the developers or if it was overlooked

Testing process:

I executed `cosmiconfig(...).search(...)` within `D:\Developer\13OnTheCode\test\index.js`

As a result, cosmiconfig recursively reached the "home directory" on the D drive, rather than the C drive

Therefore, when the configuration file is created in the root directory of the D drive, it can be found successfully

-----

I expect to have an additional handling on Windows. Based on the testing process mentioned above, when the recursive search reaches the root directory of the D drive and still cannot find the configuration file, it would be desirable to attempt searching in the user's home directory on the C drive

**Comment by d-fischer** [comment]
@13OnTheCode 

That is a completely different thing, and is intentional. If the `stopDir` is not a parent of the initial directory, the search is continued to the root and nowhere else. It's the same thing as being somewhere down in `/opt/foo/bar`, while `stopDir` defaults to the home directory, it will also search to the root and not look in the home dir. It's not a Windows specific thing.

Still, parts of this concern might be addressed by #152.

**Comment by d-fischer** [comment]
As #152 is now done and there hasn't been any more communication in this regarding the feature solving your concern, I'm closing this as duplicate.
